deepchem/data/datasets.py

```diff
@@ -2057,16 +2099,27 @@ def set_shard(self, shard_num: int, X: np.ndarray, y: Optional[np.ndarray],
     self._cached_shards = None
 
   def select(self, indices: Sequence[int],
-             select_dir: str = None) -> "DiskDataset":
+             select_dir: Optional[str] = None) -> "DiskDataset":
     """Creates a new dataset from a selection of indices from self.
 
+    Note
+    ----
+    The specified indices will be returned in sorted order. That is, if you
+    request that indices `[3, 1, 2]` are returned, you will get a
+    `DiskDataset` which contains elements in order `[1, 2, 3]`.
```

Comment:
I think this is a bug.  NumpyDataset and ImageDataset don't work this way.

Reply:
Good catch! You're right, this is a bug. It turns out that our test suite wasn't catching this. I've fixed this and will put up my new PR shortly